packages/runtime/container-runtime/src/gc/gcConfigs.ts

```diff
@@ -0,0 +1,174 @@
+		sweepEnabled = gcOptions.sweepAllowed === true;
+
+		// Set the Session Expiry only if the flag is enabled and GC is enabled.
+		if (mc.config.getBoolean(runSessionExpiryKey) && gcEnabled) {
+			sessionExpiryTimeoutMs =
+				gcOptions.sessionExpiryTimeoutMs ?? defaultSessionExpiryDurationMs;
+		}
+		sweepTimeoutMs = testOverrideSweepTimeoutMs ?? computeSweepTimeout(sessionExpiryTimeoutMs);
+
+		if (gcOptions[gcTombstoneGenerationOptionName] !== undefined) {
+			persistedGcFeatureMatrix = {
+				tombstoneGeneration: gcOptions[gcTombstoneGenerationOptionName],
+			};
+		}
+	}
+
+	/**
+	 * Whether GC should run or not. The following conditions have to be met to run sweep:
+	 *
+	 * 1. GC should be enabled for this container.
+	 *
+	 * 2. GC should not be disabled via disableGC GC option.
+	 *
+	 * These conditions can be overridden via runGCKey feature flag.
+	 */
+	const shouldRunGC =
+		mc.config.getBoolean(runGCKey) ??
+		// GC must be enabled for the document.
+		(gcEnabled &&
+			// GC must not be disabled via GC options.
+			!gcOptions.disableGC);
+
+	/**
+	 * Whether sweep should run or not. The following conditions have to be met to run sweep:
+	 *
+	 * 1. Overall GC or mark phase must be enabled (this.configs.shouldRunGC).
+	 * 2. Sweep timeout should be available. Without this, we wouldn't know when an object should be deleted.
+	 * 3. The driver must implement the policy limiting the age of snapshots used for loading. Otherwise
+	 * the Sweep Timeout calculation is not valid. We use the persisted value to ensure consistency over time.
+	 * 4. Sweep should be enabled for this container (this.sweepEnabled). This can be overridden via runSweep
+	 * feature flag.
+	 */
+	const shouldRunSweep =
+		shouldRunGC &&
+		sweepTimeoutMs !== undefined &&
+		(mc.config.getBoolean(runSweepKey) ?? sweepEnabled);
+
+	// Override inactive timeout if test config or gc options to override it is set.
+	const inactiveTimeoutMs =
+		mc.config.getNumber("Fluid.GarbageCollection.TestOverride.InactiveTimeoutMs") ??
+		gcOptions.inactiveTimeoutMs ??
+		defaultInactiveTimeoutMs;
+
+	// Inactive timeout must be greater than sweep timeout since a node goes from active -> inactive -> sweep ready.
+	if (sweepTimeoutMs !== undefined && inactiveTimeoutMs > sweepTimeoutMs) {
+		throw new UsageError("inactive timeout should not be greater than the sweep timeout");
+	}
+
+	// Whether we are running in test mode. In this mode, unreferenced nodes are immediately deleted.
+	const testMode = mc.config.getBoolean(gcTestModeKey) ?? gcOptions.runGCInTestMode === true;
+	// Whether we are running in tombstone mode. This is enabled by default if sweep won't run. It can be disabled
+	// via feature flags.
+	const tombstoneMode = !shouldRunSweep && mc.config.getBoolean(disableTombstoneKey) !== true;
+	const trackGCState = mc.config.getBoolean(trackGCStateKey) === true;
+
+	return {
+		gcEnabled,
+		sweepEnabled,
+		shouldRunGC,
+		shouldRunSweep,
+		runFullGC: gcOptions.runFullGC,
```

Comment:
super nit: for consistency, create a variable instead of inlining it.  Or inline some of the other one-liners like the 3 last ones